spatial_batch_kernels.hpp

```diff
@@ -0,0 +1,1122 @@
+/*
+ * This file is part of Vlasiator.
+ * Copyright 2010-2024 Finnish Meteorological Institute and University of Helsinki
+ *
+ * For details of usage, see the COPYING file and read the "Rules of the Road"
+ * at http://www.physics.helsinki.fi/vlasiator/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef VLASIATOR_SPATIAL_BATCH_KERNELS_HPP
+#define VLASIATOR_SPATIAL_BATCH_KERNELS_HPP
+
+#ifdef USE_WARPACCESSORS
+ #define USE_BATCH_WARPACCESSORS
+#endif
+/** GPU kernel for identifying which blocks have relevant content */
+__global__ void batch_update_velocity_block_content_lists_kernel (
+   vmesh::VelocityMesh **vmeshes,
+   vmesh::VelocityBlockContainer **blockContainers,
+   Hashinator::Hashmap<vmesh::GlobalID,vmesh::LocalID>** allMaps,
+   Real* velocity_block_min_values,
+   const bool gatherMass,
+   Real* dev_mass
+   ) {
```

Comment:
Let it be stated this once for all the times this caused me to pause today: I dislike both the closing of the ) and opening of the { at the end of the last argument and indented as such here.

Reply:
That's the auto-indentation of the clang-formatting I have in my editor. I understand your critique, and it would be possible to move the brackets on line 37 to the end of line 36, but this style looks fine to me and I do not intend to change all kernels unless there's a style guide point I am breaking.

Reply:
Move to 36 or indent to parent level, but anyway, perfectly fine and our style guide remains to be written.